code/controllers/subsystem/minimap.dm

```diff
@@ -472,15 +524,131 @@ SUBSYSTEM_DEF(minimaps)
 
 	ui = SStgui.try_update_ui(user, src, ui)
 	if(!ui)
+
+		get_current_map(user)
+		var/mob/living/carbon/xenomorph/xeno_user
+		if(isxeno(user))
+			xeno_user = user
+
+		if(ishuman(user) && skillcheck(user, SKILL_LEADERSHIP, SKILL_LEAD_EXPERT) || isqueen(user) && xeno_user.hivenumber == XENO_HIVE_NORMAL)
+			base_map_png_asset = icon2html(map_holder.map, user, sourceonly = TRUE)
+			current_menu = "home"
+		else if(isxeno(user) && xeno_user.hivenumber != XENO_HIVE_NORMAL)
+			current_menu = "view"
+		else
+			current_menu = "old_canvas"
+
 		user.client.register_map_obj(map_holder.map)
 		ui = new(user, src, "TacticalMap")
 		ui.open()
 
-/datum/tacmap/ui_static_data(mob/user)
+
+/datum/tacmap/ui_data(mob/user)
 	var/list/data = list()
+
+	// empty map dmi image used as the background for the canvas
+	data["imageSrc"] = base_map_png_asset
+
+	//flat image of most recently announced tacmap
+	data["flatImage"] = drawn_map_png_asset
+
+	data["svgData"] = svg
+
+	data["currentMenu"] = current_menu
+	data["themeId"] = theme
 	data["mapRef"] = map_holder.map_ref
+	data["toolbarColorSelection"] = toolbar_color_selection
+	data["toolbarUpdatedSelection"] = toolbar_updated_selection
+	data["worldtime"] = world.time
+	data["canvasCooldown"] = canvas_cooldown
+	data["nextCanvasTime"] = canvas_cooldown_time
+	data["updatedCanvas"] = updated_canvas
 	return data
 
+/datum/tacmap/ui_close(mob/user)
+	. = ..()
+	updated_canvas = FALSE
+	toolbar_color_selection = "black"
+	toolbar_updated_selection = "black"
+
+/datum/tacmap/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+
+	var/user = ui.user
+
+	switch (action)
+		if ("menuSelect")
+			if(params["selection"] == "old_canvas")
+				get_current_map(user)
+			current_menu = params["selection"]
+			. = TRUE
+
+		if ("clearCanvas")
+			toolbar_updated_selection = "clear"
+			. = TRUE
+		if ("updateCanvas")
+			if(alert(user, "Are you sure you want to update the canvas changes?", "Confirm?", "Yes", "No") == "No")
+				return
+			toolbar_updated_selection = "export"
+			COOLDOWN_START(src, canvas_cooldown, canvas_cooldown_time)
+
+			updated_canvas = TRUE
+			. = TRUE
+
+		if ("undoChange")
+			toolbar_updated_selection = "undo"
+			. = TRUE
+
+		if ("selectColor")
+
+			toolbar_color_selection = params["color"]
+			toolbar_updated_selection = toolbar_color_selection
+			. = TRUE
+
+		if ("selectAnnouncement")
+
+			var/icon/flat_map = getFlatIcon(map_holder.map)
```

Comment:
You were correct that the issue such as in the above image is because of this call, but I can confirm reverting #4290 allows the map to be turned into a flat icon. E.g. 
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/7f11e5ec-dbc7-4103-ae8a-8d1b366a9a29)
As far as why I'm not sure yet, but for now you can locally revert (but do not push that reversion in this PR) to test with this call working.

Even reverted though, its incorrectly coloring the icons (see how the two xenos are purpleified incorrectly compared to the normal tacmap) 
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/4a96b4a7-b782-4a75-96d5-621897d7c578)

Reply:
Additionally, the tacmap with icons should be the view when drawing so you get what you see, rather than the final result adding icons.

Reply:
Other than the color problem for the icons, instead of reverting that pr, this patch of that PR's changes will fix the regression in getFlatIcon: 
[patch.diff.txt](https://github.com/cmss13-devs/cmss13/files/12708397/patch.diff.txt)
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/19a854ec-1336-4dcc-a351-115f5a858cf5)
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/48f1c679-9c07-443c-a493-f0505a2d3a17)
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/e164bf25-19c4-4a30-8a73-c2b90ad88ca7)

Reply:
> Additionally, the tacmap with icons should be the view when drawing so you get what you see, rather than the final result adding icons.
 - This would mean I would have to flatten the icon every so often even if cic or the queen choose not to announce, I guess I could add a timer for how often that happens, but even then the flattened icon png would be out of sync with the current tacmap. Again, performance concern was the issue here, and also the issue of sync.

Reply:
> This would mean I would have to flatten the icon every so often even if cic or the queen choose not to announce, I guess I could add a timer for how often that happens, but even then the flattened icon png would be out of sync with the current tacmap. Again, performance concern was the issue here, and also the issue of sync.

It wouldn't need to be periodic - only once ever they open the new canvas.

Also leaving the view and returning back doesn't keep the drawing around - even if you update and not announce, There's no "coming back" to your drawing - or at least I certainly could not get that functionality to work in testing.

Reply:
that's what I mean, say a  so,xo,co keep switching between views, like bouncing around from the old canvas to tacmap to new canvas or whatnot, each time they open up a new canvas it would need to flatten the image to display it as the background to the canvas (the idea was once it's flattened when a member of cic opens it there is a timer preventing this sort of unnecessary accidental spam from switching between views) .That's what I wanted to avoid, but if you're saying it's not a big deal then sure, can do it that way. Maybe I'm misunderstanding - and yeah it doesn't do that, once you leave the the canvas it wipes it.

Reply:
> that's what I mean, say a so,xo,co keep switching between views, like bouncing around from the old canvas to tacmap to new canvas or whatnot, each time they open up a new canvas it would need to flatten the image to display it as the background to the canvas (the idea was once it's flattened when a member of cic opens it there is a timer preventing this sort of unnecessary accidental spam from switching between views, it will grab the original one. In a few minutes it could then be flattened again for viewing by cic ) .That's what I wanted to avoid, but if you're saying it's not a big deal then sure, can do it that way. Maybe I'm misunderstanding - and yeah it doesn't do that, once you leave the the canvas it wipes it.

What you are drawing on should be what you get. In other words, what you are drawing on should have the icons from the tacmap with people in various places. I wouldn't stress too much about a CO ending up creating a flat icon (just be sure you don't keep that old image around so it can be garbage collected). But the solution to use your cooldown lockout would be to prevent accessing that page (or it refusing to draw and update the tacmap icons) until the cooldown is complete.

I understand what you were trying to do, but even still I don't imagine it to be very intuitive if you initially open to a blank map. What you draw on should already have the icons. It shouldn't need them to opt to update the map then start drawing when it allows them to draw on a blank map.

However you could do a mix of these two things: 
1. Have it "update" the map when that page is opened.
2. Immediately have the update cooldown start.
3. Separate the announce button from the update button.
4. Consider reducing it to I don't know 2 minutes.
5. You don't really need to concern yourself with an announcement cooldown in addition to the update cooldown; but if you really want to then have at it.

Reply:
Youve already seen from discord, but just for transparency, here are the patches to get getFlatIcon to respect RESET_COLOR appearance_flags:
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/9159cfba-e142-404b-ba8d-b7db3bfa820c)

Dimension fixes + color fixes: 
[patch_all.diff.txt](https://github.com/cmss13-devs/cmss13/files/12713913/patch_all.diff.txt)
Only color fixes: 
[patch_color.diff.txt](https://github.com/cmss13-devs/cmss13/files/12713915/patch_color.diff.txt)

Do note that this means there could potentially be 2xLayer (each blip is a couple layers) more blends being performed, so this can potentially be that much more expensive of a call to perform. I still need others to evaluate what I've experimented with to see if theres anything that can be done to do it more optimally.